### PR TITLE
cafeobj 1.5.9 (new formula)

### PR DESCRIPTION
**Pull request by norbusan**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
I have tried to run this on my Linuxbrew but without success:
```
$ brew audit --strict Formula/cafeobj.rb 
==> Installing or updating 'bundler' gem
Error: Unable to resolve dependency: user requested 'did_you_mean (= 1.2.1)'
```

-----

CafeOBJ has been installed via a third party tap maintained by me in several computers already, now I would like to include it into the main Brew repository. As I am maintaining CafeOBJ and have provided (mac)ports installation options as well as Debian packages, I would like to see a brew formula, too.

I would also like to add bottles, but without a Mac computer I *think* I cannot do this.

Thanks

Norbert

### CODE REVIEW

File: Formula/cafeobj.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Cafeobj < Formula
+  desc "New generation algebraic specification and programming language"
+  homepage "https://cafeobj.org/"
+  url "http://cafeobj.org/files/1.5.9/cafeobj-1.5.9.tar.gz"
+  sha256 "a4085e9ee060a8a0a22cb7c522c17aa1ccadb5bdce8f90085e08ded3794498d4"
+  revision 1
```
Comment (lembacon) [inline Formula/cafeobj.rb]: Please remove this line.

**Comment by fxcoudert** [comment]
```
  * Stable: The URL http://cafeobj.org/files/1.5.9/cafeobj-1.5.9.tar.gz should use HTTPS rather than HTTP
  * fully scope test system calls e.g. system "#{bin}/cafeobj"
  * Emacs Lisp files were linked directly to /usr/local/share/emacs/site-lisp
    This may cause conflicts with other packages.
    They should instead be installed into:
    /usr/local/opt/cafeobj/share/emacs/site-lisp/cafeobj
    
    The offending files are:
      /usr/local/opt/cafeobj/share/emacs/site-lisp/cafeobj-mode.el
```

**Comment by norbusan** [comment]
Dear all,

I have fixed the following items:

> ```
>   * Stable: The URL http://cafeobj.org/files/1.5.9/cafeobj-1.5.9.tar.gz should use HTTPS rather than HTTP
>   * fully scope test system calls e.g. system "#{bin}/cafeobj"
> ```

But I think the check for this:
> ```
>   * Emacs Lisp files were linked directly to /usr/local/share/emacs/site-lisp
>     This may cause conflicts with other packages.
>     They should instead be installed into:
>     /usr/local/opt/cafeobj/share/emacs/site-lisp/cafeobj
> ```

is wrong. I tried here on my own system
```
$ ./configure --with-lisp=sbcl --prefix=/usr/local/
...
$ make
...
$ make install prefix=/foo/bar/baz/
...
```
and this installed the elisp file correctly into `/foo/bar/baz/share/emacs/cafeobj.el`.

I am a bit at loss what is the problem here?

Thanks for your comments

Norbert

**Comment by norbusan** [comment]
Can someone explain me how the installation is actually carried out (which arguments of make are used, which env vars) so that I can try to reproduce this?

**Comment by SMillerDev** [comment]
You could run the install with `--debug`. You can also run `brew audit` locally

**Comment by norbusan** [comment]
@SMillerDev thanks, I have now fixed the recipe and all checks have passed.

**Comment by javian** [comment]
@BrewTestBot test this please

**Comment by javian** [comment]
Just noted that the github page has very low stats https://github.com/CafeOBJ/cafeobj - is this a blocker ?

**Comment by norbusan** [comment]
> Just noted that the github page has very low stats https://github.com/CafeOBJ/cafeobj - is this a blocker ?

Just to comment on this: this project is now nearly 30 years old, was long years developed in private, then for many years on `git.cafeobj.org`, and we only recently moved to github.

In addition, as an algebraic specification language it can hardly compete with other projects ;-)

**Comment by SMillerDev** [comment]
If it passes the audit that's fine by me.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/cafeobj.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+  depends_on "sbcl"
+
+  def install
+    system "./configure", "--with-lisp=sbcl", "--prefix=#{prefix}", "--with-lispdir=#{share}/emacs/site-lisp/cafeobj"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/cafeobj", "-batch"
```
[the remark this answers is not in the document]
Comment (norbusan) [reply Formula/cafeobj.rb]: Testing whether CafeOBJ can be started and initialized. This already loads the preamble of CafeOBJ. It is a basic functionality test, that returns 0 on success.

[the remark this answers is not in the document]
Comment (fxcoudert) [reply Formula/cafeobj.rb]: Great

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @norbusan for contributing this formula to Homebrew!

**Comment by norbusan** [comment]
@fxcoudert Thanks for accepting this formula, I will keep it updated!
Thanks everyone for the feedback in improving and fixing the formula.

Thanks

Norbert

**Pull request closed without merging**